prog/heatmap_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
```

Comment:
To be honest, the test is difficult to follow. Maybe craft some same inputs manually and explicitly do the validation?